packages/smooth_app/lib/helpers/picture_capture_helper.dart

```diff
@@ -13,32 +19,66 @@ Future<bool> uploadCapturedPicture(
   required Uri imageUri,
 }) async {
   final AppLocalizations appLocalizations = AppLocalizations.of(context);
-  final SendImage image = SendImage(
-    lang: ProductQuery.getLanguage(),
+  final LocalDatabase localDatabase = context.read<LocalDatabase>();
+  final String uniqueId = _getUniqueId(imageField, barcode);
+  final BackgroundImageInputData backgroundImageInputData =
+      BackgroundImageInputData(
+    processName: IMAGE_UPLOAD_TASK,
+    uniqueId: uniqueId,
     barcode: barcode,
-    imageField: imageField,
-    imageUri: imageUri,
+    imageField: imageField.value,
+    imagePath: File(imageUri.path).path,
+    languageCode: ProductQuery.getLanguage().code,
+    user: jsonEncode(ProductQuery.getUser().toJson()),
+    country: ProductQuery.getCountry()!.iso2Code,
   );
-  final Status? result = await LoadingDialog.run<Status>(
-    context: context,
-    future: OpenFoodAPIClient.addProductImage(
-      ProductQuery.getUser(),
-      image,
+  await TaskManager().addTask(
+    Task(
+      data: backgroundImageInputData.toJson(),
+      uniqueId: uniqueId,
+    ),
+  );
+
+  localDatabase.notifyListeners();
+  // ignore: use_build_context_synchronously
+  ScaffoldMessenger.of(context).showSnackBar(
+    SnackBar(
+      content: Text(
+        appLocalizations.image_upload_queued,
+      ),
+      duration: SnackBarDuration.medium,
     ),
-    title: appLocalizations.uploading_image,
   );
-  if (result == null || result.error != null || result.status != 'status ok') {
-    await LoadingDialog.error(
-      context: context,
-      title: appLocalizations.error_occurred,
-    );
-    return false;
-  }
   //ignore: use_build_context_synchronously
   await _updateContinuousScanModel(context, barcode);
   return true;
 }
 
+/// Generates a unique id for the task , in case of tasks with the same name,
+/// it gets replaced with the new one , also for other images we randomize the id with date time so that it runs seperately
```

Comment:
please remove the useless space before the comma
+ Are you sure, you respect the 80 'characters'' limit here? You should add a line break